experimental/ActionPolyRing/src/IO.jl

```diff
@@ -0,0 +1,168 @@
+###############################################################################
+#
+#  Action polynomial rings
+#
+###############################################################################
+
+### Difference ###
+function Base.show(io::IO, ::MIME"text/plain", dpr::DifferencePolyRing)
+  io = pretty(io)
+  n = n_elementary_symbols(dpr)
+  print(io, "Difference polynomial ring in $n elementary symbols ")
+  join(io, elementary_symbols(dpr), ", ")
+  print(io, "\n")
+  print(io, "with $(ndiffs(dpr)) commuting endomorphisms\n")
+  print(io, Indent())
+  print(io, "over ", Lowercase(), base_ring(dpr))
+  print(io, Dedent())
+end
+
+function Base.show(io::IO, dpr::DifferencePolyRing)
+  io = pretty(io)
+  if is_terse(io)
+    print(io, "Difference polynomial ring")
+  else
+    print(terse(io), "Difference polynomial ring in $(n_elementary_symbols(dpr)) elementary symbols over ")
```

Comment:
```suggestion
    print(io, "Difference polynomial ring in $(n_elementary_symbols(dpr)) elementary symbols over ")
```
just the recursive call in the line below needs to be terse. this is just printing a fixed string anyway